coalib/io/Memoize.py

```diff
@@ -0,0 +1,31 @@
+import functools
+
+
+class Memoize:
```

Comment:
Actually, have you seen this? --> https://pypi.org/project/cached-property/
:sweat_smile: 

In the end you learned something though :sweat_smile: